### PR TITLE
Update the experimental annotations in the docs

### PR DESCRIPTION
**Pull request by clintongormley**
Changes in this PR:
- Removed the docs for `index.compound_format` and `index.compound_on_flush` - these are expert settings which should probably be removed (see https://github.com/elastic/elasticsearch/issues/10778)
- Removed the docs for `index.index_concurrency` - another expert setting
- Labelled the segments verbose output as experimental
- Marked the `compression`, `precision_threshold` and `rehash` options as experimental in the cardinality and percentile aggs
- Improved the experimental text on `significant_terms`, `execution_hint` in the terms agg, and `terminate_after` param on count and search
- Removed the experimental flag on the `geobounds` agg
- Marked the settings in the `merge` and `store` modules as experimental, rather than the modules themselves

### CODE REVIEW

**Comment by clintongormley** [comment]
@jpountz and @markharwood please review

File: docs/reference/search/aggregations/metrics/cardinality-aggregation.asciidoc [hunk 1]
```diff
@@ -23,10 +23,10 @@ match a query:
 
 ==== Precision control
 
-experimental[]
-
 This aggregation also supports the `precision_threshold` and `rehash` options:
 
+experimental[The `precision_threshold` and `rehash` options are specific to the current internal implementation of the `cardinality` agg, which may changein the future.]
```
Comment (jpountz) [inline docs/reference/search/aggregations/metrics/cardinality-aggregation.asciidoc]: s/changein/change in/

Comment (jpountz) [inline docs/reference/search/aggregations/metrics/cardinality-aggregation.asciidoc]: Also it's the only one where you put a dot at the end of the sentence.

**Comment by jpountz** [comment]
LGTM

**Comment by clintongormley** [comment]
@jpountz you're good ;)  pushed fixes

**Pull request closed without merging**